### PR TITLE
Add dtype functions for floating point ops

### PR DESCRIPTION
**Pull request by li-plus**
This PR is part of #1807. It added dtype functions for floating point ops that always return a tensor of dtype float32, except for input dtype float64, float16, or bfloat16.

@ramiro050 would you review this PR?

Update: the CI is crashing at `RandnDtypeDeviceModule_basic` in torchdynamo config, probably caused by the latest PyTorch as mentioned in #1792. I guess we should rebase [dtype-functions-staging](https://github.com/llvm/torch-mlir/tree/dtype-functions-staging) to the main branch to make the CI pass.

Update: rebasing on master still doesn't work. I have temporarily disabled this test in torchdynamo by `--crashing_tests_to_not_attempt_to_run_and_a_bug_is_filed RandnDtypeDeviceModule_basic`. This test case will be decomposed into a complex series of ops as shown below. Dtype functions for some ops lie in `RefineTypes` while others are in `abstract_interp_lib`. They are interleaving and failing the compilation. When the migration is finished, I will re-enable it.

```mlir
module attributes {torch.debug_module_name = "_lambda"} {
  func.func @forward() -> !torch.vtensor<[],f64> {
    %false = torch.constant.bool false
    %none = torch.constant.none
    %true = torch.constant.bool true
    %int2097152 = torch.constant.int 2097152
    %int4 = torch.constant.int 4
    %int512 = torch.constant.int 512
    %int1024 = torch.constant.int 1024
    %float1.000000e00 = torch.constant.float 1.000000e+00
    %int2097151 = torch.constant.int 2097151
    %int1 = torch.constant.int 1
    %float0.000000e00 = torch.constant.float 0.000000e+00
    %float-2.000000e00 = torch.constant.float -2.000000e+00
    %float6.283180e00 = torch.constant.float 6.283180e+00
    %int2 = torch.constant.int 2
    %int0 = torch.constant.int 0
    %cpu = torch.constant.device "cpu"
    %0 = torch.prim.ListConstruct %int4, %int512, %int1024 : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %1 = torch.aten.empty.memory_format %0, %none, %none, %cpu, %none, %none : !torch.list<int>, !torch.none, !torch.none, !torch.Device, !torch.none, !torch.none -> !torch.vtensor<[4,512,1024],f64>
    %2 = torch.aten.empty.memory_format %0, %none, %none, %cpu, %none, %none : !torch.list<int>, !torch.none, !torch.none, !torch.Device, !torch.none, !torch.none -> !torch.vtensor<[4,512,1024],f64>
    %3 = torch.aten.uniform %1, %float0.000000e00, %float1.000000e00, %none : !torch.vtensor<[4,512,1024],f64>, !torch.float, !torch.float, !torch.none -> !torch.vtensor<[4,512,1024],f64>
    %4 = torch.aten.uniform %2, %float0.000000e00, %float1.000000e00, %none : !torch.vtensor<[4,512,1024],f64>, !torch.float, !torch.float, !torch.none -> !torch.vtensor<[4,512,1024],f64>
    %5 = torch.aten.log %3 : !torch.vtensor<[4,512,1024],f64> -> !torch.vtensor<[4,512,1024],f64>
    %6 = torch.aten.mul.Scalar %5, %float-2.000000e00 : !torch.vtensor<[4,512,1024],f64>, !torch.float -> !torch.vtensor<[4,512,1024],f64>
    %7 = torch.aten.sqrt %6 : !torch.vtensor<[4,512,1024],f64> -> !torch.vtensor<[4,512,1024],f64>
    %8 = torch.aten.mul.Scalar %4, %float6.283180e00 : !torch.vtensor<[4,512,1024],f64>, !torch.float -> !torch.vtensor<[4,512,1024],f64>
    %9 = torch.aten.cos %8 : !torch.vtensor<[4,512,1024],f64> -> !torch.vtensor<[4,512,1024],f64>
    %10 = torch.aten.mul.Tensor %7, %9 : !torch.vtensor<[4,512,1024],f64>, !torch.vtensor<[4,512,1024],f64> -> !torch.vtensor<[4,512,1024],f64>
    %11 = torch.prim.ListConstruct %int0, %int1, %int2 : (!torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %12 = torch.aten.sum.dim_IntList %10, %11, %true, %none : !torch.vtensor<[4,512,1024],f64>, !torch.list<int>, !torch.bool, !torch.none -> !torch.vtensor<[1,1,1],f64>
    %13 = torch.aten.div.Scalar %12, %int2097152 : !torch.vtensor<[1,1,1],f64>, !torch.int -> !torch.vtensor<[1,1,1],f64>
    %14 = torch.aten.sub.Tensor %10, %13, %float1.000000e00 : !torch.vtensor<[4,512,1024],f64>, !torch.vtensor<[1,1,1],f64>, !torch.float -> !torch.vtensor<[4,512,1024],f64>
    %15 = torch.aten.mul.Tensor %14, %14 : !torch.vtensor<[4,512,1024],f64>, !torch.vtensor<[4,512,1024],f64> -> !torch.vtensor<[4,512,1024],f64>
    %16 = torch.aten.sum.dim_IntList %15, %11, %false, %none : !torch.vtensor<[4,512,1024],f64>, !torch.list<int>, !torch.bool, !torch.none -> !torch.vtensor<[],f64>
    torch.runtime.assert %true, "correction value should be less than or equal to productDimSize + 1"
    %17 = torch.aten.div.Scalar %16, %int2097151 : !torch.vtensor<[],f64>, !torch.int -> !torch.vtensor<[],f64>
    %18 = torch.aten.sqrt %17 : !torch.vtensor<[],f64> -> !torch.vtensor<[],f64>
    return %18 : !torch.vtensor<[],f64>
  }
}
```

### CODE REVIEW

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py [hunk 1]
```diff
@@ -1026,19 +1026,107 @@ def _get_invocations_for_op_with_tensor_arg_followed_by(*args):
...
+        Invocation(NonZeroDTensorWithDtype(torch.bool), *args),
+        Invocation(ZeroDTensorWithDtype(torch.float32), *args),
+        Invocation(ZeroDTensorWithDtype(torch.float64), *args),
+        Invocation(ZeroDTensorWithDtype(torch.bfloat16), *args),
+        Invocation(ZeroDTensorWithDtype(torch.int64), *args),
+        Invocation(ZeroDTensorWithDtype(torch.int32), *args),
+        Invocation(ZeroDTensorWithDtype(torch.bool), *args),
+    ]
+
+def _get_invocations_for_fp_only_op_with_tensor_arg_followed_by(*args):
```
Comment (ramiro050) [inline python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:1043]: I will be making a PR soon that improves the testing helper functions quite a bit, so if you're working on another set of ops, I would wait until that lands

File: build_tools/python_deploy/build_linux_packages.sh [hunk 2]
```diff
@@ -277,7 +277,7 @@ function test_in_tree() {
   python -m e2e_testing.main --config=lazy_tensor_core -v
 
   echo ":::: Run TorchDynamo e2e integration tests"
-  python -m e2e_testing.main --config=torchdynamo -v
+  python -m e2e_testing.main --config=torchdynamo -v --crashing_tests_to_not_attempt_to_run_and_a_bug_is_filed RandnDtypeDeviceModule_basic
```
Comment (ramiro050) [inline build_tools/python_deploy/build_linux_packages.sh:280]: This is fine, thanks!

**Review by ramiro050 — APPROVED** [review]
Thanks! LGTM

File: python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py (hunk 1 above)
Comment (li-plus) [reply python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:1043]: Awesome! I haven't started working on the next task. Will wait for your PR before further development.

**Pull request merged**